README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->
[![Travis-CI Build Status](https://travis-ci.org/jjchern/county.pop.svg?branch=master)](https://travis-ci.org/jjchern/county.pop) [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/jjchern/county.pop?branch=master&svg=true)](https://ci.appveyor.com/project/jjchern/county.pop)

About `county.pop`
==================

The package stores U.S. annual county-level population estimates (1969-2016) from [SEER](https://seer.cancer.gov/popdata/download.html). The dataset is unbalanced: certain counties do not have population estimates for all the years.

Installation
============

``` r
# install.packages("devtools")
devtools::install_github("jjchern/county.pop")

# To uninstall the package, use:
# remove.packages("county.pop")
```

Usage
=====

The package contains four datasets
----------------------------------

``` r
library(tibble)
county.pop::pop_since_1969
#> # A tibble: 149,765 x 15
#>    state_fips usps  state   county_fips  year   pop black other white
#>    <chr>      <chr> <chr>   <chr>       <dbl> <int> <int> <int> <int>
#>  1 01         AL    Alabama 01001        1969 23819  6879    19 16921
#>  2 01         AL    Alabama 01001        1970 24661  6914    27 17720
#>  3 01         AL    Alabama 01001        1971 25503  6948    36 18519
#>  4 01         AL    Alabama 01001        1972 27156  7196    50 19910
#>  5 01         AL    Alabama 01001        1973 28453  7352    62 21039
#>  6 01         AL    Alabama 01001        1974 29261  7344    75 21842
#>  7 01         AL    Alabama 01001        1975 29716  7294    83 22339
#>  8 01         AL    Alabama 01001        1976 29892  7153    96 22643
#>  9 01         AL    Alabama 01001        1977 30457  7145   107 23205
#> 10 01         AL    Alabama 01001        1978 30879  7089   118 23672
#> # ... with 149,755 more rows, and 6 more variables: age_1_18 <int>,
#> #   age_19_44 <int>, age_45_64 <int>, age_65_85p <int>, female <int>,
#> #   male <int>

county.pop::pop_since_1969_adjusted
#> # A tibble: 149,766 x 15
#>    state_fips usps  state   county_fips  year   pop black other white
#>    <chr>      <chr> <chr>   <chr>       <dbl> <int> <int> <int> <int>
#>  1 01         AL    Alabama 01001        1969 23819  6879    19 16921
#>  2 01         AL    Alabama 01001        1970 24661  6914    27 17720
#>  3 01         AL    Alabama 01001        1971 25503  6948    36 18519
#>  4 01         AL    Alabama 01001        1972 27156  7196    50 19910
#>  5 01         AL    Alabama 01001        1973 28453  7352    62 21039
#>  6 01         AL    Alabama 01001        1974 29261  7344    75 21842
#>  7 01         AL    Alabama 01001        1975 29716  7294    83 22339
#>  8 01         AL    Alabama 01001        1976 29892  7153    96 22643
#>  9 01         AL    Alabama 01001        1977 30457  7145   107 23205
#> 10 01         AL    Alabama 01001        1978 30879  7089   118 23672
#> # ... with 149,756 more rows, and 6 more variables: age_1_18 <int>,
#> #   age_19_44 <int>, age_45_64 <int>, age_65_85p <int>, female <int>,
#> #   male <int>

county.pop::pop_since_1990
#> # A tibble: 84,718 x 16
#>    state_fips usps  state  county_fips  year   pop  hisp  nh_b  nh_o  nh_w
#>    <chr>      <chr> <chr>  <chr>       <dbl> <int> <int> <int> <int> <int>
#>  1 01         AL    Alaba… 01001        1990 34353   213  6817   189 27134
#>  2 01         AL    Alaba… 01001        1991 35010   227  6823   204 27756
#>  3 01         AL    Alaba… 01001        1992 35985   245  6920   225 28595
#>  4 01         AL    Alaba… 01001        1993 36953   264  7017   241 29431
#>  5 01         AL    Alaba… 01001        1994 38186   305  7155   262 30464
#>  6 01         AL    Alaba… 01001        1995 39112   332  7242   282 31256
#>  7 01         AL    Alaba… 01001        1996 40207   382  7319   308 32198
#>  8 01         AL    Alaba… 01001        1997 41238   439  7415   335 33049
#>  9 01         AL    Alaba… 01001        1998 42106   479  7451   353 33823
#> 10 01         AL    Alaba… 01001        1999 42963   550  7498   384 34531
#> # ... with 84,708 more rows, and 6 more variables: age_1_18 <int>,
#> #   age_19_44 <int>, age_45_64 <int>, age_65_85p <int>, female <int>,
#> #   male <int>

county.pop::pop_since_1990_adjusted
#> # A tibble: 84,719 x 16
#>    state_fips usps  state  county_fips  year   pop  hisp  nh_b  nh_o  nh_w
#>    <chr>      <chr> <chr>  <chr>       <dbl> <int> <int> <int> <int> <int>
#>  1 01         AL    Alaba… 01001        1990 34353   213  6817   189 27134
#>  2 01         AL    Alaba… 01001        1991 35010   227  6823   204 27756
#>  3 01         AL    Alaba… 01001        1992 35985   245  6920   225 28595
#>  4 01         AL    Alaba… 01001        1993 36953   264  7017   241 29431
#>  5 01         AL    Alaba… 01001        1994 38186   305  7155   262 30464
#>  6 01         AL    Alaba… 01001        1995 39112   332  7242   282 31256
#>  7 01         AL    Alaba… 01001        1996 40207   382  7319   308 32198
#>  8 01         AL    Alaba… 01001        1997 41238   439  7415   335 33049
#>  9 01         AL    Alaba… 01001        1998 42106   479  7451   353 33823
#> 10 01         AL    Alaba… 01001        1999 42963   550  7498   384 34531
#> # ... with 84,709 more rows, and 6 more variables: age_1_18 <int>,
#> #   age_19_44 <int>, age_45_64 <int>, age_65_85p <int>, female <int>,
#> #   male <int>
```

The population counts are usually based on July 1 populations. The two adjusted versions take into account the changes in population due to hurricanes Katrina and Rita.

All four data sets provide total population counts at the county-year level, as well as population counts by

-   race (and ethnicity):
    -   `pop_since_1969`: black, other, white
    -   `pop_since_1990`: Hispanic, non-Hispanic black, non-Hispanic other, and non-Hispanic white
-   age groups (0-18, 19-44, 45-64, 65-85+), and
-   gender (female and male).

New and Deleted Counties are Documented Properly
------------------------------------------------

For example, Broomfield County, Colorado (08014) are created on November 15, 2001 (<https://www.census.gov/geo/reference/county-changes.html>).

``` r
library(tidyverse)
#> ── Attaching packages ────────────────────────────────────────── tidyverse 1.2.1 ──
#> ✔ ggplot2 2.2.1          ✔ purrr   0.2.4     
#> ✔ tidyr   0.8.0          ✔ dplyr   0.7.4     
#> ✔ readr   1.1.1.9000     ✔ stringr 1.2.0     
#> ✔ ggplot2 2.2.1          ✔ forcats 0.2.0
#> ── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
#> ✖ dplyr::filter() masks stats::filter()
#> ✖ dplyr::lag()    masks stats::lag()
county.pop::pop_since_1990 %>% 
    filter(county_fips == "08014") %>% 
    distinct(year)
#> # A tibble: 15 x 1
#>     year
#>    <dbl>
#>  1  2002
#>  2  2003
#>  3  2004
#>  4  2005
#>  5  2006
#>  6  2007
#>  7  2008
#>  8  2009
#>  9  2010
#> 10  2011
#> 11  2012
#> 12  2013
#> 13  2014
#> 14  2015
#> 15  2016
```

Demographic Shifts
------------------

<img src="man/figures/README-age_65p-1.png" width="100%" />


NEWS.md

# county.pop 0.0.4

* Add both unadjusted and adjusted version of raw files
* Add population counts by race, gender, and age groups

# county.pop 0.0.3

* Another Trivial patch

# county.pop 0.0.2

* Trivial patch

# county.pop 0.0.1

* Included county-year level population estimates, 1969-2016
* Added a `NEWS.md` file to track changes to the package.


R/data.R

#' @title County-Level Population, 1969-2016, Adjusted
#'
#' @md
#' @description County-level population from SEER, adjusted for the population shifts due to hurricanes Katrina (August 29) and Rita (September 24).
#'
#' @source [https://seer.cancer.gov/popdata/download.html]()
"pop_since_1969_adjusted"

#' @title County-Level Population, 1969-2016
#'
#' @md
#' @description County-level population from SEER
#'
#' @source [https://seer.cancer.gov/popdata/download.html]()
"pop_since_1969"

#' @title County-Level Population, 1990-2016, Adjusted
#'
#' @md
#' @description County-level population from SEER, adjusted for the population shifts due to hurricanes Katrina (August 29) and Rita (September 24).
#'
#' @source [https://seer.cancer.gov/popdata/download.html]()
"pop_since_1990_adjusted"

#' @title County-Level Population, 1990-2016
#'
#' @md
#' @description County-level population from SEER
#'
#' @source [https://seer.cancer.gov/popdata/download.html]()
"pop_since_1990"


data-raw/prep.R

library(tidyverse)

url = "https://seer.cancer.gov/popdata/yr1969_2016.19ages/us.1969_2016.19ages.adjusted.txt.gz"
fil = "data-raw/us.1969_2016.19ages.adjusted.txt.gz"
if (!exists(fil)) download.file(url, fil)

# Load raw data -----------------------------------------------------------

read_fwf("data-raw/us.1969_2016.19ages.adjusted.txt",
         fwf_widths(c(4, 2, 2, 3, 2, 1, 1, 1, 2, 8),
                    c("year", "usps", "state_fips", "county_fips", "registry",
                      "race", "origin", "sex", "age", "pop"))) %>%
    print() -> raw

# Aggregate to county-year level ------------------------------------------

raw %>%
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    arrange(state_fips, usps, county_fips, year) %>%
    group_by(state_fips, usps, county_fips, year) %>%
    summarise(pop = pop %>% sum()) %>%
    ungroup() %>%
    print() -> pop

devtools::use_data(pop, overwrite = TRUE)


data-raw/prep_since_1969.R

# dict: https://seer.cancer.gov/popdata/popdic.html -----------------------

library(tidyverse)

# Option 1: Use the 19 age groups
# url = "https://seer.cancer.gov/popdata/yr1969_2016.19ages/us.1969_2016.19ages.txt.gz"
# fil_gz = "data-raw/us.1969_2016.19ages.txt.gz"
# fil    = "data-raw/us.1969_2016.19ages.txt"

# Option 2: Use the single year age groups
url = "https://seer.cancer.gov/popdata/yr1969_2016.singleages/us.1969_2016.singleages.txt.gz"
fil_gz = "data-raw/us.1969_2016.singleages.txt.gz"
fil    = "data-raw/us.1969_2016.singleages.txt"

if (!file.exists(fil_gz)) {
    download.file(url, fil_gz)
    R.utils::gunzip(fil_gz, fil, remove = FALSE)
}

# Load raw data -----------------------------------------------------------

read_fwf(fil,
         fwf_widths(c(4, 2, 2, 3, 2, 1, 1, 1, 2, 8),
                    c("year", "usps", "state_fips", "county_fips", "registry",
                      "race", "origin", "sex", "age", "pop"))) %>%
    print() -> raw

raw %>% sample_n(1000) %>% distinct(race)
raw %>% sample_n(1000) %>% distinct(origin)
raw %>% sample_n(1000) %>% distinct(sex)
raw %>% sample_n(1000) %>% distinct(registry)
raw %>% sample_n(1000) %>% distinct(age)

# Aggregate to county-year level ------------------------------------------

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    arrange(state_fips, usps, county_fips, year) %>%
    group_by(state_fips, usps, county_fips, year) %>%
    summarise(pop = pop %>% sum()) %>%
    ungroup() %>%
    print() -> pop

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(race = case_when(
        race == 1 ~ "white",
        race == 2 ~ "black",
        race == 3 ~ "other"
    )) %>%
    group_by(state_fips, usps, county_fips, year, race) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(race, pop) %>%
    print() -> pop_by_race

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(age = age %>% as.integer()) %>%
    mutate(age_grp = case_when(
        age %in% 0:18  ~ "age_1_18",
        age %in% 19:44 ~ "age_19_44",
        age %in% 45:64 ~ "age_45_64",
        age %in% 65:85 ~ "age_65_85p"
    )) %>%
    group_by(state_fips, usps, county_fips, year, age_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(age_grp, pop) %>%
    print() -> pop_by_age_grp

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(sex_grp = if_else(sex == 2, "female", "male")) %>%
    group_by(state_fips, usps, county_fips, year, sex_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(sex_grp, pop) %>%
    print() -> pop_by_sex

pop %>%
    left_join(fips::state) %>%
    select(state_fips, usps, state, county_fips, year, pop) %>%
    left_join(pop_by_race) %>%
    left_join(pop_by_age_grp) %>%
    left_join(pop_by_sex) %>%
    print() -> pop_since_1969

devtools::use_data(pop_since_1969, overwrite = TRUE)


data-raw/prep_since_1990.R

# dict: https://seer.cancer.gov/popdata/popdic.html -----------------------

library(tidyverse)

# Option 1: Use the 19 age groups
# url = "https://seer.cancer.gov/popdata/yr1990_2016.19ages/us.1990_2016.19ages.txt.gz"
# fil_gz = "data-raw/us.1990_2016.19ages.txt.gz"
# fil    = "data-raw/us.1990_2016.19ages.txt"

# Option 2: Use the single year age groups
url = "https://seer.cancer.gov/popdata/yr1990_2016.singleages/us.1990_2016.singleages.txt.gz"
fil_gz = "data-raw/us.1990_2016.singleages.txt.gz"
fil    = "data-raw/us.1990_2016.singleages.txt"

if (!file.exists(fil_gz)) {
    download.file(url, fil_gz)
    R.utils::gunzip(fil_gz, fil, remove = FALSE)
}

# Load raw data -----------------------------------------------------------

read_fwf(fil,
         fwf_widths(c(4, 2, 2, 3, 2, 1, 1, 1, 2, 8),
                    c("year", "usps", "state_fips", "county_fips", "registry",
                      "race", "origin", "sex", "age", "pop"))) %>%
    print() -> raw

raw %>% sample_n(1000) %>% distinct(race)
raw %>% sample_n(1000) %>% distinct(origin)
raw %>% sample_n(1000) %>% distinct(sex)
raw %>% sample_n(1000) %>% distinct(registry)
raw %>% sample_n(1000) %>% distinct(age)

# Aggregate to county-year level population --------------------------------

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    arrange(state_fips, usps, county_fips, year) %>%
    group_by(state_fips, usps, county_fips, year) %>%
    summarise(pop = pop %>% sum()) %>%
    ungroup() %>%
    print() -> pop

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(race_ethncy = case_when(
        origin == 0 & race == 1         ~ "nh_w", # Non-Hispanic White
        origin == 0 & race == 2         ~ "nh_b", # Non-Hispanic Black
        origin == 0 & race %in% c(3, 4) ~ "nh_o", # Non-Hispanic Others
        origin == 1                     ~ "hisp"  # Hispanic
    )) %>%
    group_by(state_fips, usps, county_fips, year, race_ethncy) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(race_ethncy, pop) %>%
    print() -> pop_by_race

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(age = age %>% as.integer()) %>%
    mutate(age_grp = case_when(
        age %in% 0:18  ~ "age_1_18",
        age %in% 19:44 ~ "age_19_44",
        age %in% 45:64 ~ "age_45_64",
        age %in% 65:85 ~ "age_65_85p"
    )) %>%
    group_by(state_fips, usps, county_fips, year, age_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(age_grp, pop) %>%
    print() -> pop_by_age_grp

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(sex_grp = if_else(sex == 2, "female", "male")) %>%
    group_by(state_fips, usps, county_fips, year, sex_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(sex_grp, pop) %>%
    print() -> pop_by_sex

pop %>%
    left_join(fips::state) %>%
    select(state_fips, usps, state, county_fips, year, pop) %>%
    left_join(pop_by_race) %>%
    left_join(pop_by_age_grp) %>%
    left_join(pop_by_sex) %>%
    print() -> pop_since_1990

devtools::use_data(pop_since_1990, overwrite = TRUE)


data-raw/prep_since_1990_adjusted.R

# dict: https://seer.cancer.gov/popdata/popdic.html -----------------------

library(tidyverse)

# Option 1: Use the 19 age groups
# url = "https://seer.cancer.gov/popdata/yr1990_2016.19ages/us.1990_2016.19ages.adjusted.txt.gz"
# fil_gz = "data-raw/us.1990_2016.19ages.adjusted.txt.gz"
# fil    = "data-raw/us.1990_2016.19ages.adjusted.txt"

# Option 2: Use the single year age groups
url = "https://seer.cancer.gov/popdata/yr1990_2016.singleages/us.1990_2016.singleages.adjusted.txt.gz"
fil_gz = "data-raw/us.1990_2016.singleages.adjusted.txt.gz"
fil    = "data-raw/us.1990_2016.singleages.adjusted.txt"

if (!file.exists(fil_gz)) {
    download.file(url, fil_gz)
    R.utils::gunzip(fil_gz, fil, remove = FALSE)
}

# Load raw data -----------------------------------------------------------

read_fwf(fil,
         fwf_widths(c(4, 2, 2, 3, 2, 1, 1, 1, 2, 8),
                    c("year", "usps", "state_fips", "county_fips", "registry",
                      "race", "origin", "sex", "age", "pop"))) %>%
    print() -> raw

raw %>% sample_n(1000) %>% distinct(race)
raw %>% sample_n(1000) %>% distinct(origin)
raw %>% sample_n(1000) %>% distinct(sex)
raw %>% sample_n(1000) %>% distinct(registry)
raw %>% sample_n(1000) %>% distinct(age)

# Aggregate to county-year level population --------------------------------

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    arrange(state_fips, usps, county_fips, year) %>%
    group_by(state_fips, usps, county_fips, year) %>%
    summarise(pop = pop %>% sum()) %>%
    ungroup() %>%
    print() -> pop

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(race_ethncy = case_when(
        origin == 0 & race == 1         ~ "nh_w", # Non-Hispanic White
        origin == 0 & race == 2         ~ "nh_b", # Non-Hispanic Black
        origin == 0 & race %in% c(3, 4) ~ "nh_o", # Non-Hispanic Others
        origin == 1                     ~ "hisp"  # Hispanic
    )) %>%
    group_by(state_fips, usps, county_fips, year, race_ethncy) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(race_ethncy, pop) %>%
    print() -> pop_by_race

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(age = age %>% as.integer()) %>%
    mutate(age_grp = case_when(
        age %in% 0:18  ~ "age_1_18",
        age %in% 19:44 ~ "age_19_44",
        age %in% 45:64 ~ "age_45_64",
        age %in% 65:85 ~ "age_65_85p"
    )) %>%
    group_by(state_fips, usps, county_fips, year, age_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(age_grp, pop) %>%
    print() -> pop_by_age_grp

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(sex_grp = if_else(sex == 2, "female", "male")) %>%
    group_by(state_fips, usps, county_fips, year, sex_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(sex_grp, pop) %>%
    print() -> pop_by_sex

pop %>%
    left_join(fips::state) %>%
    select(state_fips, usps, state, county_fips, year, pop) %>%
    left_join(pop_by_race) %>%
    left_join(pop_by_age_grp) %>%
    left_join(pop_by_sex) %>%
    print() -> pop_since_1990_adjusted

devtools::use_data(pop_since_1990_adjusted, overwrite = TRUE)


data-raw/prep_since_1969_adjusted.R

# dict: https://seer.cancer.gov/popdata/popdic.html -----------------------

library(tidyverse)

# Option 1: Use the 19 age groups
# url = "https://seer.cancer.gov/popdata/yr1969_2016.19ages/us.1969_2016.19ages.adjusted.txt.gz"
# fil_gz = "data-raw/us.1969_2016.19ages.adjusted.txt.gz"
# fil    = "data-raw/us.1969_2016.19ages.adjusted.txt"

# Option 2: Use the single year age groups
url = "https://seer.cancer.gov/popdata/yr1969_2016.singleages/us.1969_2016.singleages.adjusted.txt.gz"
fil_gz = "data-raw/us.1969_2016.singleages.adjusted.txt.gz"
fil    = "data-raw/us.1969_2016.singleages.adjusted.txt"

if (!file.exists(fil_gz)) {
    download.file(url, fil_gz)
    R.utils::gunzip(fil_gz, fil, remove = FALSE)
}

# Load raw data -----------------------------------------------------------

read_fwf(fil,
         fwf_widths(c(4, 2, 2, 3, 2, 1, 1, 1, 2, 8),
                    c("year", "usps", "state_fips", "county_fips", "registry",
                      "race", "origin", "sex", "age", "pop"))) %>%
    print() -> raw

raw %>% sample_n(1000) %>% distinct(race)
raw %>% sample_n(1000) %>% distinct(origin)
raw %>% sample_n(1000) %>% distinct(sex)
raw %>% sample_n(1000) %>% distinct(registry)
raw %>% sample_n(1000) %>% distinct(age)

# Aggregate to county-year level ------------------------------------------

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    arrange(state_fips, usps, county_fips, year) %>%
    group_by(state_fips, usps, county_fips, year) %>%
    summarise(pop = pop %>% sum()) %>%
    ungroup() %>%
    print() -> pop

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(race = case_when(
        race == 1 ~ "white",
        race == 2 ~ "black",
        race == 3 ~ "other"
    )) %>%
    group_by(state_fips, usps, county_fips, year, race) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(race, pop) %>%
    print() -> pop_by_race

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(age = age %>% as.integer()) %>%
    mutate(age_grp = case_when(
        age %in% 0:18  ~ "age_1_18",
        age %in% 19:44 ~ "age_19_44",
        age %in% 45:64 ~ "age_45_64",
        age %in% 65:85 ~ "age_65_85p"
    )) %>%
    group_by(state_fips, usps, county_fips, year, age_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(age_grp, pop) %>%
    print() -> pop_by_age_grp

raw %>%
    # sample_n(100) %>% # Speed up the pipe
    mutate(county_fips = paste0(state_fips, county_fips)) %>%
    mutate(pop = pop %>% as.integer()) %>%
    mutate(sex_grp = if_else(sex == 2, "female", "male")) %>%
    group_by(state_fips, usps, county_fips, year, sex_grp) %>%
    summarise(pop = pop %>% sum()) %>%
    spread(sex_grp, pop) %>%
    print() -> pop_by_sex

pop %>%
    left_join(fips::state) %>%
    select(state_fips, usps, state, county_fips, year, pop) %>%
    left_join(pop_by_race) %>%
    left_join(pop_by_age_grp) %>%
    left_join(pop_by_sex) %>%
    print() -> pop_since_1969_adjusted

devtools::use_data(pop_since_1969_adjusted, overwrite = TRUE)
